src/Nethermind/Nethermind.Blockchain/Find/LogFinder.cs

```diff
@@ -66,12 +69,22 @@ private bool ShouldUseBloomDatabase(BlockHeader fromBlock, BlockHeader toBlock)
 
         private IEnumerable<FilterLog> FilterLogsWithBloomsIndex(LogFilter filter, BlockHeader fromBlock, BlockHeader toBlock)
         {
+            Block FindBlock(long blockNumber)
+            {
+                var block = _blockFinder.FindBlock(blockNumber);
+                if (block == null)
+                {
+                    if (_logger.IsError) _logger.Error($"Could not find block {blockNumber} in database. eth_getLogs will return incomplete results.");
```

Comment:
Should we return error to RPC?

Reply:
hmmm, you can be smart and guess which one is on main by analyzing parents... but this seems like a hassle

Reply:
It can mean analyzing millions of blocks - bad idea.